src/include/libpmemfile-core.h

```diff
@@ -60,12 +60,19 @@ void pmemfile_pool_close(PMEMfilepool *pfp);
 
 PMEMfile *pmemfile_open(PMEMfilepool *pfp, const char *pathname, int flags,
 		...);
+PMEMfile *pmemfile_openat(PMEMfilepool *pfp, PMEMfile *file,
```

Comment:
If pmemfile-core will use file descriptors, some applications will use the same variable for storing pmem and non-pmem file handles (with additional flag telling about the type). So the code will look like:
```
if (pmem_fd)
   pmemfile_write(pfp, fd, buf, len);
else
   write(fd, buf, len);
```
If some parts of the code base will not be updated, this might lead to bugs.

There are syscalls that will succeed for these descriptors, silently corrupting data/state.
E.g.
- "write" will do nothing
- "close" will close the descriptor, without releasing any pmemfile resources; next open may get the same file descriptor, which pmemfile will see as a collision
- "dup" will duplicate /dev/null descriptor
- "fstat" will return info for /dev/null

Forcing users to have different container for pmemfile file handles solves these problems.

And anyone sane enough will abstract file operations to the pmemfile-core-based backend, which will nullify any advantage of using file descriptors.